### DISCUSSION TITLE
fips 140-2 support

### DISCUSSION THREAD

**Discussion by esnyderCPI** [Hakky54/ayza#225] [Q&A]
Hello,

   I've been looking for X509 support with pem files and I ran across your project.  The only piece I'm now missing is fips 140-2 support.  Is there a way to leverage your library along with fips?  I do not see anything referring to fips support.

Thanks

**Comment by Hakky54** [comment]
Hi,

Thank you for discovering this library. Fips 140 is needed for application/libraries which actually implements cryptographic algorithms. This library is not implementing cryptographic algorithms. It provides convenient methods for constructing objects like TrustManager, KeyManager SSLContext etc. The actual implementation of cryptographic algorithms is provided by the JDK itself when you just use this library alogside with the JDK. So Fips 140-2 is not applicable for this library.

What I would suggest is to use a fips validated provider, for example `bouncy castle-fips` alongside with this library and that should work for your use case.
